CityOfDetroit/bloom
# 1337/homepage coming soon

## Issue

Closes #1337 

## Description

This PR adds the 'Coming Soon' section to Detroit's public homepage. This section is designed to show three listings that are marked as coming soon and having the closest opening date appear first. It also includes a 'See All Coming Soon' button that takes the user to the filtered page. This button is visible even when there are 3 or less listings marked as Coming Soon to encourage the user to go to the listings page and potentially explore other filters.

Update: This PR also refactors the Regions section to no longer use the Horizontal Scroll component and rather follow a similar pattern that the Coming Soon section does. NOTE: in order to have the region images scale dynamically while remaining the same relative size, I used the aspect-ratio which is a newer css property. From my testing and their documentation, it is supported on every browser expect for Internet Explorer. On the off chance that a user has an outdated browser that doesn't support this property, the only difference is that the region images would be slightly taller overall and could show gently different heights in certain mid-sized screen sizes. These differences would be difficult to notice and the homepage remains presentable either way.

## How Can This Be Tested/Reviewed?

This can be tested and reviewed by going to the Detroit Public homepage viewing the section in different screen dimensions.  Then click the See All Coming Soon button and verify that you land on the listings page with the Coming Soon filter set to true. Additionally, go the partner's side and vary the amount of public listings marked as Coming Soon and their publishing dates to see the affect on the home page.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
